components/camera/videosource/align.go

```diff
@@ -1,278 +0,0 @@
-			if err == nil {
-				err = warp.CheckValid()
-			}
-			return warp, err
-		})
-
-	config.RegisterComponentAttributeConverter(camera.SubtypeName, "align_color_depth", "intrinsic_extrinsic",
-		func(val interface{}) (interface{}, error) {
-			b, err := json.Marshal(val)
-			if err != nil {
-				return nil, err
-			}
-			matrices, err := transform.NewDepthColorIntrinsicsExtrinsicsFromBytes(b)
-			if err != nil {
-				return nil, err
-			}
-			err = matrices.CheckValid()
-			return matrices, err
-		})
-
-	config.RegisterComponentAttributeConverter(camera.SubtypeName, "align_color_depth", "homography",
-		func(val interface{}) (interface{}, error) {
-			homography := &transform.RawDepthColorHomography{}
-			decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{TagName: "json", Result: homography})
-			if err != nil {
-				return nil, err
-			}
-			err = decoder.Decode(val)
-			if err == nil {
-				err = homography.CheckValid()
-			}
-			return homography, err
-		})
-}
-
-func getAligner(attrs *alignAttrs, logger golog.Logger) (transform.Aligner, error) {
-	switch {
-	case attrs.IntrinsicExtrinsic != nil:
-		cam, ok := attrs.IntrinsicExtrinsic.(*transform.DepthColorIntrinsicsExtrinsics)
-		if !ok {
-			return nil, rdkutils.NewUnexpectedTypeError(cam, attrs.IntrinsicExtrinsic)
-		}
-		return cam, nil
-	case attrs.Homography != nil:
-		conf, ok := attrs.Homography.(*transform.RawDepthColorHomography)
-		if !ok {
-			return nil, rdkutils.NewUnexpectedTypeError(conf, attrs.Homography)
-		}
-		cam, err := transform.NewDepthColorHomography(conf)
-		if err != nil {
-			return nil, err
-		}
-		return cam, nil
-	case attrs.Warp != nil:
-		conf, ok := attrs.Warp.(*transform.AlignConfig)
-		if !ok {
-			return nil, rdkutils.NewUnexpectedTypeError(conf, attrs.Warp)
-		}
-		cam, err := transform.NewDepthColorWarpTransforms(conf, logger)
-		if err != nil {
-			return nil, err
-		}
-		return cam, nil
-	default:
-		return nil, nil
-	}
-}
-
-// alignAttrs is the attribute struct for aligning.
-type alignAttrs struct {
-	CameraParameters     *transform.PinholeCameraIntrinsics `json:"intrinsic_parameters,omitempty"`
-	DistortionParameters *transform.BrownConrady            `json:"distortion_parameters,omitempty"`
-	Stream               string                             `json:"stream"`
-	Debug                bool                               `json:"debug,omitempty"`
-	Color                string                             `json:"color_camera_name"`
-	Depth                string                             `json:"depth_camera_name"`
-	Height               int                                `json:"height_px"`
-	Width                int                                `json:"width_px"`
-	IntrinsicExtrinsic   interface{}                        `json:"intrinsic_extrinsic,omitempty"`
-	Homography           interface{}                        `json:"homography,omitempty"`
-	Warp                 interface{}                        `json:"warp,omitempty"`
```

Comment:
Was `Warp` completely removed?

Reply:
Yes, Warp has been superseded by the `homography`. Warp was an earlier model that had fewer degrees of freedom than homography, in that it did not encode "skew" parameters. homography is more general and can do anything "warp" did.